### ISSUE TITLE
Describe how to use with kubernetes

### ISSUE THREAD

**Issue by fabiand** [kubevirt/must-gather#28]
Hey,

please describe in the readme how I can use msut gather with kubernetes

**Comment by pkliczewski** [comment]
@fabiand must-gather is openshift specific. We could provide manual steps to collect diagnostic data based on [collection scripts](https://github.com/kubevirt/must-gather/tree/master/collection-scripts).

**Comment by fabiand** [comment]
But can't we run the container image against k8s as well?

- fabian

**Comment by pkliczewski** [comment]
Container image uses `openshift-must-gather` [inspect](https://github.com/openshift/must-gather/blob/master/pkg/cmd/inspect/inspect.go) which is openshift specific. We would need to implement something similar in `kubectl` or `virtctl`.

**Comment by masayag** [comment]
> Container image uses `openshift-must-gather` [inspect](https://github.com/openshift/must-gather/blob/master/pkg/cmd/inspect/inspect.go) which is openshift specific. We would need to implement something similar in `kubectl` or `virtctl`.

Or use the OKD as the provider. @fabiand is that acceptable to require using OKD in order to enhance must-gather ?

**Comment by fabiand** [comment]
I'm not sure I understand, Moti. Can you rephrase?

**Comment by masayag** [comment]
I assumed since the code of the base image openshift-must-gather makes strong assumption about the existence of cluster-operators, but after I tried running it against k8s, I noticed differently. The cluster-operator resource are removed from the gather script, hence, no failures on that part.

However, the current image is designed to run on openshift + hco. There are namespaces assumed to appear. This can be mitigated by checking the existence of the namespaces or cr before including them in the resource list to collect.

```
Error: namespaces "kubevirt-hyperconverged" not found
Error: namespaces "openshift-operator-lifecycle-manager" not found
Error: namespaces "openshift-marketplace" not found
Error: namespaces "kubemacpool-system" not found
Error: namespaces "linux-bridge" not found
Error: namespaces "openshift-sdn" not found
Error: namespaces "nmstate" not found
Error: namespaces "sriov" not found
Error: namespaces "kubevirt-web-ui" not found
Error: the server doesn't have a resource type "v2vvmwares"
Error: namespaces "cdi" not found
Error: the server doesn't have a resource type "datavolumes"
Error: the server doesn't have a resource type "nodenetworkstates"
Error: the server doesn't have a resource type "istag"
error: the server doesn't have a resource type "templates"
```
Other than that - the instruction for k8s cluster are the same as described on the 'README.me'.

I'm closing this issue and reporting a new one for missing ns/resources.
